Reconstruction/RecCalorimeter/src/components/SingleCaloTowerTool.cpp

```diff
@@ -0,0 +1,144 @@
+#include "SingleCaloTowerTool.h"
+
+// FCCSW
+#include "DetInterface/IGeoSvc.h"
+
+// datamodel
+#include "datamodel/CaloHitCollection.h"
+#include "datamodel/CaloHit.h"
+
+// DD4hep
+#include "DD4hep/LCDD.h"
+#include "DD4hep/Readout.h"
+
+DECLARE_TOOL_FACTORY(SingleCaloTowerTool)
+
+SingleCaloTowerTool::SingleCaloTowerTool(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<ITowerTool>(this);
+  declareInput("cells", m_cells, "calo/cells");
+  declareProperty("readoutName", m_readoutName);
+  // the default value to calculate the position of clusters
+  declareProperty("radiusForPosition", m_radius = 1.);
+  declareProperty("etaMax", m_etaMax = 0.);
+  declareProperty("deltaEtaTower", m_deltaEtaTower = 0.01);
+  declareProperty("deltaPhiTower", m_deltaPhiTower = 0.01);
+  // needed for AlgTool wit output/input until it appears in Gaudi AlgTool constructor
+  declareProperty("DataInputs", inputDataObjects());
+  declareProperty("DataOutputs", outputDataObjects());
+}
+
+StatusCode SingleCaloTowerTool::initialize() {
+  if (GaudiTool::initialize().isFailure()) {
+    return StatusCode::FAILURE;
+  }
+  m_geoSvc = service("GeoSvc");
+  if (!m_geoSvc) {
+    error() << "Unable to locate Geometry Service. "
+            << "Make sure you have GeoSvc and SimSvc in the right order in the configuration." << endmsg;
+    return StatusCode::FAILURE;
+  }
+  // check if readouts exist
+  if (m_geoSvc->lcdd()->readouts().find(m_readoutName) == m_geoSvc->lcdd()->readouts().end()) {
+    error() << "Readout <<" << m_readoutName << ">> does not exist." << endmsg;
+    return StatusCode::FAILURE;
+  }
+  // retrieve PhiEta segmentation
+  m_segmentation = dynamic_cast<DD4hep::DDSegmentation::GridPhiEta*>(
+      m_geoSvc->lcdd()->readout(m_readoutName).segmentation().segmentation());
+  if (m_segmentation == nullptr) {
+    error() << "There is no phi-eta segmentation." << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if( ! m_etaMax) {
+    warning() << "Undefined detector size in eta. In each event the cell collection will be searched for maximum eta." << endmsg;
+  }
+  return StatusCode::SUCCESS;
+}
+
+StatusCode SingleCaloTowerTool::finalize() { return GaudiTool::finalize(); }
+
+std::array<int, 2> SingleCaloTowerTool::towersNumber() {
+  // number of phi bins
+  m_nPhiTower = idPhi(2 * M_PI);
+  // number of eta bins (if eta maximum is defined)
+  if( m_etaMax) {
+    m_nEtaTower = 2 * idEta(m_etaMax - m_deltaEtaTower / 2.) + 1;
+  } else {
+    m_nEtaTower = 0;
+  }
+  return {m_nEtaTower, m_nPhiTower};
+}
+
+int SingleCaloTowerTool::etaTowersNumber() {
+  // recalculate number of eta bins for each event
+  const fcc::CaloHitCollection* cells = m_cells.get();
+  float etaCell = 0;
+  for (const auto& cell : *cells) {
+    etaCell = fabs(m_segmentation->eta(cell.core().cellId));
+    if( etaCell > m_etaMax ) {
+      m_etaMax = etaCell;
```

Comment:
It is overridden if it is equal to 0 (= default). I added it also to the doxy comment.